http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java

```diff
@@ -124,6 +154,13 @@ public static SdkEventLoopGroup create(EventLoopGroup eventLoopGroup, ChannelFac
      * <p>
      * {@link ChannelFactory} will be resolved based on the type of {@link EventLoopGroup} provided. IllegalArgumentException will
      * be thrown for any unknown EventLoopGroup type.
+     * <p>
+     * If {@link MultiThreadIoEventLoopGroup} is passed in, {@link NioSocketChannel} and {@link NioDatagramChannel} will be
+     * resolved, regardless of the transport {@link IoHandlerFactory} set on the {@link MultiThreadIoEventLoopGroup}. This is
+     * because it is not possible to determine the type of transport factory from a given {@link MultiThreadIoEventLoopGroup}.
+     * <p>
+     * To use a {@link MultiThreadIoEventLoopGroup} with a non-Nio transport factory, use
+     * {@link #create(EventLoopGroup, ChannelFactory, ChannelFactory)}, specifying the socket and datagram channels.
```

Comment:
The way this is worded is hard to follow. From what I understand, if the customer calls this with a `MultiThreadIoEventLoopGroup` (and not one of its subclasses like `NioEventLoopGroup`), then the SDK will disregard the `IoHandler` the event loop group was constructed with when choosing the factory, and always assume `NioSocketChannel::new`.

And if they want to ensure that specific factories are used, to use one of the other overloads?

What happens if there is a mismatch in the `IoHandlerFactory` and the resolved `SocketFactory` (e.g. epoll vs NIO)? Is there a clear error message?